app/policies/comment_policy.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class CommentPolicy < ApplicationPolicy
+  private
+
+  def comment
+    @record
+  end
+
+  def user_required?
+    false
+  end
+
+  def task_authorized?
+    if @record.respond_to?(:map)
+      @record.map(&:tasks).uniq.all? {|task| Pundit.policy(@user, task).show? }
+    else
+      Pundit.policy(@user, comment.task).show?
+    end
+  end
+
+  public
```

Comment:
Is there any specific reason you started with the private methods above and then switched again to public? Normally, it's just the other way around in Ruby (omitting `public`).

Reply:
I wanted to declare the helper methods before the actual authorization because I figured it was more readable. Since the policy is not that long anyway I switched the order like you suggested.